### PR TITLE
Apply robust harvest effect to current plant only and not offspring

### PR DESCRIPTION
**Pull request by PolterTzi**
## About the PR

Reworked the effects of Robust Harvest to be more temporary.
Now it gives a boost to the potency of the current plant that is no longer inherited by any offspring.
The boost is additive with the plant's base potency and any mutation it has, no longer limiting the combined potency.
Currently the maximum boost is 45.
The yield decrease from overdose has been made to affect YieldMod instead, making it also temporary.
The seedless effect from Robust Harvest has been removed.

## Why / Balance

Arguably more realistic than just fertilising once and making all subsequent plants give overgrown crops. 
The seedless trait is AFAIK largely meant to prevent the easy production of powerful plants, since this makes the resulting seeds no longer as potent, I hope it's fair to no longer use it for fertiliser.

This would make it somewhat easier to avoid getting seedless plants, at least for the less experimentative botanists, and keep plant "lineages" alive.
Conversely, it adds both a resource and work cost to keeping the plants potent by needing each new plant to be independently fertilised.
Hopefully that's a fair trade-off.
It also makes the effectiveness of fertiliser and mutations no longer dependent on each other, making it .

## Technical details

Adds new fields to plantholdercomponent and producecomponent to store the potency boost and the potency of the produce.
The boost is added to the base potency of a plant when harvesting.
Robust Harvest now only increases the potency boost.
YeildMod has been replaced with a float for easier modification.

## Media

The bottom tray has had 15u of Robust Harvest added to it.
image
image
image
image
image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

The functions responsible for creating produce in the BotanySystem have been changed to take potency boost as an argument.
The name of Potency in SeedData was changed to BasePotency, reflecting the "inherited" potency, rather than increase caused by RH. If this breaks something, change the reference to the variable to whatever is appropriate in your scope.
YeildMod has been replaced with a float.

**Changelog**

:cl:
- tweak: Robust Harvest no longer makes a plant's seeds more potent, but stacks independently with mutations for up to 45 additional potency at 15 units.
- tweak: Robust Harvest no longer makes plants seedless.

### CODE REVIEW

**Comment by gitjubx** [comment]
*shakes head*....you will make it even harder since you anyway will hit seedless wall ...once you try to mutate potency up or crossbreed....you would nerv botany even harder , aslong with the cliping change so idk just do it because everyone will lose interest in even the simplest form of botany ..honestl idk where you get those...stupid ideas ...

**Comment by GentleButter** [comment]
why you gatta keep ruining botany bruv

**Comment by Tayrtahn** [comment]
> also whats arguably more realistic about this? I farm IRL and if you genetically modify a crop its seeds inherit the genetic modification smdh

Isn't Robust Harvest just fertilizer though? Feeding a plant shouldn't genetically modify it. My SS14 botany knowledge is limited, but isn't that supposed to be the role of unstable mutagen or Left 4 Zed?

**Comment by slarticodefast** [comment]
If I have understood your code changes correctly, robust harvest will still reduce the yield, which is genetic, while only giving a temporary potency bonus.

Robust harvest is capped at 50 potency. For higher values mutation is needed, so I do not think it is overpowered.

The yield decrease is already there to balance the higher amount of reagents. The total amount of reagents per harvest is only increased slightly. The main difference is that much less grinding is needed. I often see chemistry not use the medicinal plants botany provides because grinding them up would be too much work. With this PR this would become even worse.

You would need a huge amount of robust harvest to produce a reasonable amount of higher potency harvest. I don't think it is worth it for chemistry to make that only for a minor harvest reagent increase, when you could just harvest more plants instead.

All in all I think botany has recently been nerfed enough with the clipping changes. I used to play botanist a lot, but at the moment it is not so much fun and I generally see a lack of botany players. I hope this will change with the upcoming addition of the plant analyzer.

**Comment by PolterTzi** [comment]
> If I have understood your code changes correctly, robust harvest will still reduce the yield, which is genetic, while only giving a temporary potency bonus.
> 
> Robust harvest is capped at 50 potency. For higher values mutation is needed, so I do not think it is overpowered.

If I understand the code right, the yield decrease is only when you add too much of the reagent and it still metabolises while the plant is maxed out.
Though I should probably rewrite that part as well, since it would be odd that one effect is temporary while the other is permanent.

**Comment by slarticodefast** [comment]
> If I understand the code right, the yield decrease is only when you add too much of the reagent and it still metabolises while the plant is maxed out.

Oh, you are right. I always added 15u of RH, which is enough for increasing potency to 50 and sometimes reduced the yield depending on the base potency (somewhere from 1-25). But I think both changes either being temporary or permanent would be more consistent. Otherwise clipping yield would decrease over time.

**Comment by K-Dynamic** [comment]
Yeah robust harvest caps out potency at 50, and going above requires mutations that we _could_ do

The thing is mutations are unpredictable and rather hell to keep track of (especially without plant scanner still); I personally wouldn't try mutating for higher potency beyond 50, let for lineage to retain any increased potency at all

Just consider the fact you're swapping out a consistent mechanic for more RNG, and getting mutagen itself can be rng depending on availability and mood of chemists

**Comment by PolterTzi** [comment]
> why you gatta keep ruining botany bruv

~~It makes me feel powerful.~~

I'll be honest, I don't have the best design sense.

**Comment by gitjubx** [comment]
> All in all I think botany has recently been nerfed enough with the clipping changes. I used to play botanist a lot, but at the moment it is not so much fun and I generally see a lack of botany players. I hope this will change with the upcoming addition of the plant analyzer.

since the clipping stuff came like half of the good botanist quit because it was no longer fun and removed lots of possibilitys you could do as botanist , it was overall terrible designt with seed health and stage 2 restriction since mostly kudzu destroys it befor you can clip or other issues while mutating or even keeping the most basic linage alive forcing you to either use tons of dylo , or grow tons of your own fertilizer plants in addition to hoarding seeds .....like if it would just consume a lot of water and nutriment from tray if you clip instead the other stuff...you atleast have to constantly also refillt trays with stuff but arent bound to tideous healing the plant over and over and keep clipping

**Comment by slarticodefast** [comment]
So I did some math and testing for some common ingredients. In the following table I calculated the ingredient gain from 15u of RH which gives you a 45 potency bonus. This costs chemistry 5 nitrogen, phosphorus and potassium each to make.

name | chemical | potency divisor | amount at base potency | amount increase from RH
--- | --- | ---: | ---: | ---:
wheat | flour | 20 | 5.25u | 2.5u
cannabis | THC | 10 | 3u | 4.5u
ambrosia deus | omnizine | 35 | 1.29u | 1.19u
galaxy thistle | stellibinin | 4 | 3.5u | 11.25u
fly amanita | amatoxin | 4 | 3.5u | 11.25u
koibean | carpotoxin | 30 | 1.17u | 1.5u

In many cases is is easier and cheaper to simply harvest another plant than to use the new RH. I do not think it is worth it for chemistry to make RH for botany under these conditions. Potency would have to be completely rebalanced around this change.

You might also want to wait with balancing until [this PR](https://github.com/space-wizards/space-station-14/pull/25575) is merged since it changes how potency works.

Another problem I see is that the potency change is no longer directly communicated to the player. Previously this was done by clipping the plant and checking the seed. For temporary changes this is not possible.

Personally I would prefer if this change was not merged, even if balanced. While it would be more realistic, it is ok to sacrifice some realism for game design reasons.

**Comment by PolterTzi** [comment]
@[slarticodefast](https://github.com/slarticodefast)
You make a good point. It should be balanced to be worthwhile to use.
I think I'm conceptually on a worthwhile track with this change, but it does indeed need balancing.

**Comment by Jezithyr** [comment]
Reopened since this author's repoban was in error. (The ban was made due to a misunderstood reference to the stonewall riots in an april fools PR)

**Comment by mehx3** [comment]
As a botanist main I don't get why people are mad at this change. After the clipping changes you couldn't use Robust anyway, if you were doing mutation and you want to be able to have the ability to get seeds from the mutated plant. Honestly, i wish seedless was completely removed, and yield was capped to 3 on default seeds and capped to 5 trough mutations. This would incentivize getting higher potency, instead of spamming plants, but this a different topic, probably. I wonder though - if you mutate plant to 44 potency, add Robust to get 89 "temporary" potency, and mutate potency again. Would it jump to 86/100 potency or 30/58 potency?

**Comment by gitjubx** [comment]
> As a botanist main I don't get why people are mad at this change. After the clipping changes you couldn't use Robust anyway, if you were doing mutation and you want to be able to have the ability to get seeds from the mutated plant. Honestly, i wish seedless was completely removed, and yield was capped to 3 on default seeds and capped to 5 trough mutations. This would incentivize getting higher potency, instead of spamming plants, but this a different topic, probably. I wonder though - if you mutate plant to 44 potency, add Robust to get 89 "temporary" potency, and mutate potency again. Would it jump to 86/100 potency or 30/58 potency?

you clearly arent a main botanist , because else you would just heal the robusted plant to get more clippings via nutri/vita chemicals or dylo/cryo if the chemist supplies you...there was always a reason to use robust like just for ghalaxthistel and keep making 1-3 plants at 50 potency and clip and repeat since you got the vitamins/nutriment to keep the linage up while also getting more seeds.....and to your other question you would have 58 or 30 if it downgrades but i think if you have 44 it doesnt skip the 50 so it would just upgrade to 50 then afterwards 58 if another roll

**Comment by mehx3** [comment]
> you clearly arent a main botanist , because else you would just heal the robusted plant to get more clippings via nutri/vita chemicals or dylo/cryo if the chemist supplies you...there was always a reason to use robust like just for ghalaxthistel and keep making 1-3 plants at 50 potency and clip and repeat since you got the vitamins/nutriment to keep the linage up while also getting more seeds.....and to your other question you would have 58 or 30 if it downgrades but i think if you have 44 it doesnt skip the 50 so it would just upgrade to 50 then afterwards 58 if another roll

I'll tell you more - you can use sterile swabs to completely heal your plant, don't even need to use chems.
If you have cooperative chemist you can always ask for a bucket of Robust which, when you add roundstart supplies, would be enough for a lot of plants. Also you can always get ressuply kits for your NutriMax. On topic of nutriments/vitamins - 1-2 robusted lemon/cocoa plant would give you more than enough produce to heal your plants. Personally, I mostly use Left-4-Zed with mutagen which, most of the time, mutates default plants potency to 58-72, making robust completely useless anyway. Creating some type of "economy" to Robust is good IMO, also it incentivizes getting "multiple harvests" mutation, so i like this PR. And going back to my original question - it skips 50, you can't get 50 trough mutation, and did you test it or just making a guess?

**Comment by gitjubx** [comment]
i heard once aswell with a trick you can use sterile swabs endlessly to crossbreed ...i am not sure if you can reset health each time with swabs this would be new to me , but i guess since it would also negate most stuff from the clipping change ( no efford to heal just swap and done) would get nerved quickly,  left4zed takes time and rng to stack the mutation multiplier its pure rng to end up with a 72 straight away i could also end up with a 72 if i just added 5u mutagen and luck , you dont normaly use left4zed just to get more potency...more to stack chemicals mutation rolls and other but as all with botany its just pure rng and gambling... i didnt tested i just spontan guessed i forgot its a - + 14 from potency rolls? got confused why from 50 its just +8 on a proc but afterwards its 72 - 86...so yea your right it would be the second one  the 30/58 seed potency and product potency from added robust with 89 wouldt prob be 100 or 86 if it downgrades but not sure there since i cant test it with the robust change

**Comment by PolterTzi** [comment]
> I wonder though - if you mutate plant to 44 potency, add Robust to get 89 "temporary" potency, and mutate potency again. Would it jump to 86/100 potency or 30/58 potency?

The bonus should be mutually independent with any mutations.
So if you take a 44 potency plant, add 15u robust, its crops should have the equivalent potency of 89, the "genetics" of the plant still have 44 potency.
When the plant mutates, the genetic potency is modified, say it decreases by 14.
You would have a plant whose seeds have potency of 30 and whose crops have the potency of 75.

**Comment by mehx3** [comment]
> Well after i was told about this a lot and teased...i just wanna leave a comment here to describe what i think of this... Foul-smelling Egg.... Its not a good concept to change the basic robust to be something ..more confusing to the new botanist with temporary effects and some more factors. The argument to change robust to counter the seedless trait somehow doesnt matter , since once you swap 1 time or mutate its gonna be seedless most of the time anyway and i rather prefer to have a none rng way to get a 50 potency plant rather than having to ask the chemist not only for mutagen but also for more robust each time ... if you would make a new chemical in addition to robust maybe something that could increase temporar yield or temporar potency in exchange for production time or something else....there it would have some potentional

Making RH harder to understand for newbie botanist is a bad thing, i can agree. But i disagree that making so it doesn't give seedless is pointless. If you are doing mutations using pure mutagen, getting seedless is a relatively rare occurrence. And furthermore, if you are doing mutations, you shouldn't even add RH in the first place since you can always add it as a last step in case you actually need it. Using swabs gives you seedless 70% of the time, and because of it you can try and get interesting plants through mutation without using swabs at all, so you can easily get seeds without having to grow/ask for healing chems. Making this temporary effect into different chem would just make it never used, since you would have to specifically ask for it, as you mentioned as a downside to this change. In my ~300hr in botany I haven't asked for Robust ONCE and i think that's not great. All in all I believe that this change would bring more useful things, than it would take away(PotencyDiv shenanigans can be fixed with simple numbers changing).

**Comment by PolterTzi** [comment]
I don't think the effect being temporary is strictly more confusing for new players. It's as simple as "fertiliser doesn't affect seeds". It's only really an issue if you're used to the current behaviour.
And, yeah, the current values aren't balanced for this approach, so the numbers would need to be tweaked for this to work.
I don't think asking chem for robust should be an issue, just gotta balance it so the rate of consumption compared to output is reasonable.

I know this is kinda a casual take, but you can do a reasonable amount of botany work without too many mutations or crossbreeding, I would argue the change (if properly balanced) would make "basic" botany somewhat more approachable as opposed to dealing with making a plant seedless by just adding a reasonable amount of fertiliser.

**Comment by mehx3** [comment]
> if you do mutations you do it AFTER using RH in most cases since you can get extra potency procs to 58 or 72 if your lucky or not

If you start with default plant and add mutagen in a controlled manner, i.e. 5u at a time, and potency mutates - you get 44 potency. If you start with robusted plant and add mutagen the same way and potency mutates - you get to 58 potency with guaranteed seedless. 14 point of potency is not worth it IMO. Getting 72 potency from 5u of mutagen is extremely rare(less than 1%) and you might as well use L4Z, mutagen and phalanximine to easily get there from any plant. By adding RH before mutations you basically just skip 1 step in potency ladder and at the same time doom yourself to seedless. Doing mutations while trying not to get seedless is a viable method and this PR would add the option to use RH if you are doing mutations this way.

**Comment by mehx3** [comment]
> L4Z (- !type:PlantAdjustMutationMod prob: 0.3 amount: 0.4) so 1u has a 30% chance to increase modifier by 0.4 up to 3 for 3 rolls instead of 1 from mutagen from what my understanding is so using L4Z you also gamble with the amount you use from like 5u up to like 25u to reach the full potention i dont know how much L4Z you are using and in combination with other chems like dylo/weedkiller but ...i wouldnt use left4zed just for potency alone in mind but its good to see people using it.

If you add 10u of L4Z you would end up with 2.2 MutationMod on average. Then, if you add 25u of mutagen you would get 5 rolls with 11 MutationLevel(2.2*5). This will make it so you would have 5 rolls with 40% chance to change potency(it's ~11% with 5 MutationLevel rolls). All those rolls would most likely bring you to 58/72 potency because those numbers are in the middle of the potency mutation ladder(i.e. if you would roll potency mutation 1000 times you would almost definitely end up on 58/72). This method also gives you a bunch of random gases and chemicals, and seedless as well, sadly, but using L4Z is not the way to avoid getting seedless. But I still believe that there is use in trying not to get seedless. Honestly seedless is kind of a shitty mechanic since you get it for doing anything, there is no way to get rid of it, and it can be relatively easily played around at the same time. Also going above 30 potency only triggers seedless if you are using RH.

File: Content.Server/Botany/SeedPrototype.cs
Comment (Partmedia) [inline Content.Server/Botany/SeedPrototype.cs:179]: Please don't rename this. Rather, update the comment documenting what this variable now does and how it's different from before.

File: Content.Server/Botany/Components/PlantHolderComponent.cs
Comment (Partmedia) [inline Content.Server/Botany/Components/PlantHolderComponent.cs:75]: Please document why this is here and how it is different from `Potency` in SeedData.

File: Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs
Comment (Partmedia) [inline Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs:45]: Guard against divide by zero.

File: Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs
Comment (Partmedia) [inline Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs:41]: This cast is unnecessary.

File: Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs
Comment (Partmedia) [inline Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs:45]: This seems like it reduces all the yield in one step. Would it be better to gradually reduce yield and increase YieldReductionProbability?

File: Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs
Comment (Partmedia) [inline Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs:23]: Extraneous space

File: Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs
Comment (Partmedia) [inline Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs:21]: Please document what this does

File: Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs
Comment (Partmedia) [inline Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs:15]: Please document what this now does

File: Content.Server/Botany/Systems/BotanySystem.Seed.cs
Comment (Partmedia) [inline Content.Server/Botany/Systems/BotanySystem.Seed.cs:152]: Should this be zero?

File: Content.Server/Botany/Systems/BotanySystem.Seed.cs
Comment (Partmedia) [inline Content.Server/Botany/Systems/BotanySystem.Seed.cs:154]: Remove space after cast.

**Review by Partmedia — CHANGES_REQUESTED** [review]

File: Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs
Comment (PolterTzi) [reply Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs:45]: Ok, I'll admit, this is kinda hacky.
This is supposed to reduce the yieldmod by the equivalent of a single produce.
So if a plant has 5 yield, it should reduce it by 0.2, thus reducing the amount harvested by 1.

File: Content.Server/Botany/Systems/BotanySystem.Seed.cs
Comment (PolterTzi) [reply Content.Server/Botany/Systems/BotanySystem.Seed.cs:152]: Seemed to me that if yieldmod ever were to reach into the negatives (which should not be possible in current behaviour, but still), the resulting yield should end up as zero. Or one.
The original code seemed more like, uh, a "panic" fallback than properly handling negative values.

File: Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs
Comment (Partmedia) [reply Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs:45]: This sounds fine when explained. Go ahead and add a comment explaining what the intended behavior is (which is to decrease yield by one plant unit).

**Comment by PolterTzi** [comment]
...I didn't expect the @ to do that, please excuse me, random person I apparently accidentally mentioned.

File: Content.Server/Botany/Systems/BotanySystem.Seed.cs
Comment (Partmedia) [reply Content.Server/Botany/Systems/BotanySystem.Seed.cs:154]: This was not resolved.

Comment (PolterTzi) [reply Content.Server/Botany/Systems/BotanySystem.Seed.cs:154]: I though putting spaces after cast was the appropriate formatting.
My IDE complains when there aren't spaces.

Comment (Partmedia) [reply Content.Server/Botany/Systems/BotanySystem.Seed.cs:154]: Your IDE is not authoritative with respect to style guidelines. If you are using a style configuration file that we provided, please file a bug reporting the inconsistency.

The [official C# documentation](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/types/casting-and-type-conversions) shows that there should be no space after the cast.

Comment (PolterTzi) [reply Content.Server/Botany/Systems/BotanySystem.Seed.cs:154]: Huh, really did not expect the IDE to be wrong like that.
Will fix those then.

**Comment by mirrorcult** [comment]
im iffy on balancing changes like this that arent part of a more holistic botany doc, not confident numerous small changes that people are generally unhappy with are going to solve any big issues

**Comment by mirrorcult** [comment]
why are you doomering like "i guess the fact that the majority doesnt like it doesnt matter!!!" when the only maint comment indicative of anything is me being unconvinced its a good idea

**Comment by PolterTzi** [comment]
> wouldnt merge this how it is and actually just make another chemical for it , mayority is against it but i guess that says nothing =) but the aftereffects will

At this point, I'll defer to whatever the maintainer decision ends up being. (not like I have many other options, really)
Making it into a new chem is certainly an interesting idea, but I feel like the current RobustHarvest parallel with something like this could make for a less than balanced combination.

**Comment by PolterTzi** [comment]
> not dooming anything i will only say you will see afterwards whats gonna happen because i just wouldnt touch the basic stuff and rather add stuff onto it ( there are more comments than just you, and there was also some talk about it just not here ) just check the like/dislike to get an idea and its not taken into account the other talk thats not here about it ( i once said it would be good to add this maybe as extra chemical rather than to change how robust is working , so i am not fully against it )

I don't imagine this would entirely turn my case around, but I'd like to note the PR started off I would say worse than it is now, so while I certainly can't blame all of the negative reactions on that, I feel like that may have played a role.

**Comment by PolterTzi** [comment]
> yea i mean of course it would need some balance like to prevent getting to high potency to quickly but to use it as alternative if you want tempo potency but also be able to extract seeds

It would certainly be an interesting alternative, though the issue would be to have it be both powerful enough to be viable and not be too powerful stacked with RH.

**Comment by PolterTzi** [comment]
> yea it is better from what it started thats true but its still..well a mayor change for most botanist that isnt took well and maybe its aswell what you did befor that influenced these things now ( i mean not what you did but like how you did it)

I'd like to note, for the record, while the clipping nerf has my name on it, I only made it in response to and as a replacement for a previous nerf,. (and for that matter, based on someone else's idea)

File: Content.Server/Botany/Components/PlantHolderComponent.cs
Comment (Partmedia) [inline Content.Server/Botany/Components/PlantHolderComponent.cs:77]: Since this datafield matches the automatically-derived name, the explicit DataField("potencyBonus") can be omitted. I know the code above and below still use the old convention, but let's try to update the convention in the new code.

**Comment by Partmedia** [comment]
Hi @PolterTzi , thanks for your original and continued work on this. I felt like this was a reasonable change to consider for Robust Harvest that address some of the balance issues currently around this. But seeing that folks are quite split on how mechanics in botany might work, I think this is a good candidate to design doc out and come to a decision on during the freeze.

I'm going to close this PR now. During the freeze there will be an opportunity for everyone who has a stake in botany to chip in on putting together a design doc should hopefully flesh out how botany should work in depth, and after the freeze we can decide whether or not to go forward for this.

To that end, please keep your branch available so that we can resurrect this after the freeze should we decide that this is the best way to move forward.

Thanks again for your work so far and sorry this didn't work out now.

**Comment by PolterTzi** [comment]
@Partmedia 
Would it be appropriate now to ask for re-opening this PR?
Or should I try writing docs first and try to "push" this once those are finalised?

**Comment by Partmedia** [comment]
There you go.

**Comment by PolterTzi** [comment]
Alright, this now has merge conflicts, was made ages ago and I haven't really kept up with the developments in the community.
I guess, unless and until stated otherwise, I'll just close this PR.
I'll leave the branch around, so the code itself shouldn't get lost if someone (me or otherwise) tries to revive this ancient curse.

**Pull request closed without merging**
